core/src/main/scala/kafka/server/BrokerMetadataCheckpoint.scala

```diff
@@ -102,11 +112,16 @@ class RawMetaProperties(val props: Properties = new Properties()) {
 }
 
 object MetaProperties {
+  def apply(clusterId: String, nodeId: Int): MetaProperties = {
+    MetaProperties(clusterId, nodeId, MetadataVersion.latest().version())
+  }
+
   def parse(properties: RawMetaProperties): MetaProperties = {
     properties.requireVersion(expectedVersion = 1)
     val clusterId = require(ClusterIdKey, properties.clusterId)
     val nodeId = require(NodeIdKey, properties.nodeId)
-    new MetaProperties(clusterId, nodeId)
+    val metadataVersion = properties.initialMetadataVersion.getOrElse(MetadataVersion.stable().version())
```

Comment:
Should this be MetadataVersion.V1? is this the case of the metadata version before 3.2.0?

Reply:
You're right, this should be V1. This covers the case of an existing meta.properties file that does not have the initial metadata.version specified. We're using this as a signal that a cluster was on a KRaft preview version previously.